moonraker/components/notifier.py

```diff
@@ -74,6 +84,36 @@ async def _on_job_state_changed(
         for notifier in self.events.get(evt_name, []):
             await notifier.notify(evt_name, [prev_stats, new_stats])
 
+    async def _on_job_layer_changed(
+            self,
+            job_event: JobEvent,
+            prev_stats: Dict[str, Any],
+            new_stats: Dict[str, Any]
+    ) -> None:
+        evt_name = str(job_event)
+        if "info" in new_stats:
+            for notifier in self.events.get(evt_name, []):
+                # get config value:
+                # percentage of layers to generate the layer change notification
+                layer_trigger = notifier.layer_trigger
+                # get config value:
+                # only send message if the layer number is greater than this value.
+                minimum_layers = notifier.minimum_layers
+                # total layers from the gcode
+                total = int(new_stats["info"]["total_layer"])
+                # current layer counter
+                current = int(new_stats["info"]["current_layer"])
+                # calculate if we've hit a % trigger point above the min height
+                # this isn't nice to read because of the 88 char line limit
+                if layer_trigger != 0 and total >= minimum_layers and current > 0 and \
+                    current != total and \
+                    round(current/total/layer_trigger, 5)//1 != \
+                        round(((current-1)/total)/layer_trigger, 5)//1:
+
+                    logging.info(f"Layer change notification at \
+                    {round(current/total/layer_trigger,5)//1*layer_trigger*100}%")
```

Comment:
Continuing over a new line with a backslash includes the indentation as actual characters, producing spaced out output like `Layer change notification at                     25.0%`.
```suggestion
                    logging.info(
                        "Layer change notification at "
                        f"{round(current/total/layer_trigger,5)//1*layer_trigger*100}%"
                    )
```

Reply:
Thanks for the code review and feedback. I wasn't used to restricting line lengths, but now I know :). I'll commit these changes.